src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -531,6 +531,12 @@ private IEnumerable<IDiagnostic> GatherParameterMismatchDiagnostics(ISemanticMod
 
             // emit diagnostic only if there is a using statement
             var usingDeclarationSyntax = this.Root.UsingDeclarationSyntax;
+
+            if (usingDeclarationSyntax is not null && usingDeclarationSyntax.Path is NoneLiteralSyntax)
+            {
+                yield break;
+            }
```

Comment:
It's better to move this to the top of the method since there is not need to calculate `missingAssignedParams` and `missingRequriedParams` when there is `using none`.

Reply:
thanks for the comment, done 👍